shelter/remoteattestation/ratlsclient.c

```diff
@@ -63,6 +63,10 @@ int ra_tls_echo(int sockfd, enclave_tls_log_level_t log_level,
 		len = sizeof(buf) - 1;
 	buf[len] = '\0';
 	printf("Server:\n%s\n", buf);
+	*recemsglen = len;
+	for (int i = 0; i < len; i++) {
+		retmsg[i] = buf[i];
+	}
```

Comment:
Why not use retmsg directly as the parameter of enclave_tls_receive()?

Reply:
Good question! Retmsg will not bring the content to golang context.
I try to use retmsg=buf, it will failed to bring data to golang context.

Reply:
I don't see what the essential differences between your codes and this sample code:
```go
package main

/*
#include <string.h>

void test(unsigned char *a)
{
        memset(a, 14, 16);
}
*/
import "C"
import (
        "unsafe"
        "fmt"
)

func main() {
        var arr [16]byte;

        C.test((*C.uchar)((unsafe.Pointer(&arr[0]))))
        if arr[3] != 14 {
                fmt.Println("wrong")
        }
}
```

I think retmsg can be used directly by enclave_tls_receive().

Reply:
You both guys are correct! Sorry for my misoperation on parse retmsg into json error by add '\0' then i wrongly think the pointer usage can't work. 
Correct it as you guys mentioned method. Thx!